easypy/concurrency.py

```diff
@@ -955,6 +958,77 @@ def _get_my_ident():
     return Hex(threading.current_thread().ident)
 
 
+class LoggedRLock():
+    """
+    Like RLock, but more logging friendly.
+
+        name: give it a name, so it's identifiable in the logs
+        log_interval: the interval between log messages
+        default_expiration: throw an exception if the lock is held for more than this duration
+    """
+
+    # we could inherit from this and support other types, but that'll require changes in the repr
+    LockType = threading.RLock
+
+    slots = ("_lock", "_name", "_default_expiration", "_timer", "_log_interval")
+    _RE_OWNER = re.compile(".*owner=(\d+) count=(\d+).*")
+
+    def __init__(self, name=None, log_interval=15, default_expiration=None):
+        self._lock = self.__class__.LockType()
+        self._name = name or '{}-{:X}'.format(self.LockType.__name__, id(self))
+        self._default_expiration = default_expiration
+        self._expiration_timer = None
+        self._log_interval = log_interval
+
+        # we want to support both the gevent and builtin lock
+        try:
+            self._lock._owner
+        except AttributeError:
+            def _get_data():
+                return tuple(map(int, self._RE_OWNER.match(repr(self._lock)).groups()))
+        else:
+            def _get_data():
+                return self._lock._owner, self._lock._count
+        self._get_data = _get_data
+
+    def __repr__(self):
+        owner, count = self._get_data()
+        try:
+            owner = threading._active[owner].name
+        except KeyError:
+            pass
+        if owner:
+            return "<{}: owned by <{}>x{} for {}>".format(self._name, owner, count, self._expiration_timer.elapsed)
+        else:
+            return "<{}: unowned>".format(self._name)
+
+    def acquire(self, blocking=True, timeout=-1, expiration=None):
```

Comment:
Why both blocking and timeout?  
Is not timeout=None == non blocking and timeout explicitly blocking?